cli/src/main.rs

```diff
@@ -4,98 +4,90 @@
-		)
-		.subcommand(SubCommand::with_name("repl").about("Starts a JavaScript Shell"))
-		.subcommand(
-			SubCommand::with_name("run")
-				.about("Runs a JavaScript File")
-				.arg(
-					Arg::with_name("path")
-						.help("The JavaScript file to be run. Default: 'main.js'")
-						.required(false),
-				)
-				.arg(
-					Arg::with_name("log_level")
-						.help("Sets Logging Level. Default: ERROR")
-						.takes_value(true)
-						.short("l")
-						.long("loglevel")
-						.required(false)
-						.conflicts_with("debug"),
-				)
-				.arg(
-					Arg::with_name("debug")
-						.help("Sets Logging Level to DEBUG.")
-						.short("d")
-						.long("debug")
-						.required(false),
-				)
-				.arg(
-					Arg::with_name("script")
-						.help("Disables ES Modules Features")
-						.long("script")
-						.required(false),
-				),
-		)
-		.get_matches();
+#[derive(StructOpt)]
+#[structopt(name = "spiderfire", about = "JavaScript Runtime")]
+struct Cli {
+	#[structopt(subcommand)]
+	commands: Option<Commands>,
+}
 
-	match matches.subcommand_name() {
-		Some("eval") => {
-			let subcmd = matches.subcommand_matches("eval").unwrap();
+#[derive(StructOpt)]
+pub enum Commands {
+	#[structopt(about = "Evaluates a line of JavaScript")]
+	Eval {
+		#[structopt(required(true), about = "Line of JavaScript to be evaluated")]
+		source: String,
+	},
 
+	#[structopt(about = "Starts a JavaScript Shell")]
+	Repl,
+
+	#[structopt(about = "Runs a JavaScript file")]
+	Run {
+		#[structopt(about = "The JavaScript file to run. Default: 'main.js'", required(false), default_value = "main.js")]
+		path: String,
+
+		#[structopt(about = "Sets logging level, Default: ERROR", short, long, required(false), default_value = "error")]
+		log_level: String,
+
+		#[structopt(about = "Sets logging level to DEBUG.", short, long)]
+		debug: bool,
+
+		#[structopt(about = "Disables ES Modules Features", short, long)]
+		script: bool,
+	},
+}
+
+fn main() {
+	let args = Cli::from_args();
+
+	match args.commands {
+		Some(Eval { source }) => {
 			CONFIG
 				.set(Config::default().log_level(LogLevel::Debug).script(true))
 				.expect("Config Initialisation Failed");
-			eval::eval_source(subcmd.value_of("source").unwrap());
-		}
-		Some("repl") => {
-			CONFIG
-				.set(Config::default().log_level(LogLevel::Debug).script(true))
-				.expect("Config Initialisation Failed");
-			repl::start_repl();
+			eval::eval_source(source);
 		}
-		Some("run") => {
-			let subcmd = matches.subcommand_matches("run").unwrap();
 
-			let log_level = if subcmd.is_present("debug") {
-				LogLevel::Debug
-			} else if let Some(level) = subcmd.value_of("log_level") {
-				let level = level.to_uppercase();
-				match level.as_str() {
-					"NONE" => LogLevel::None,
-					"INFO" => LogLevel::Info,
-					"WARN" => LogLevel::Warn,
-					"ERROR" => LogLevel::Error,
-					"DEBUG" => LogLevel::Debug,
+		Some(Run {
+			path,
+			log_level,
+			debug,
+			script,
+		}) => {
+			let mut log_lev = LogLevel::Error;
```

Comment:
I think this can be converted to something more similar to the old method so that we don't need to initialise a mutable variable here.
(Leave the `log_level.to_uppercase().to_str()`, that reduces it by one line)

https://github.com/Redfire75369/spiderfire/blob/df18729a3be3ec780d2bbd48c549f4ac3b6e8130/cli/src/main.rs#L78-L92